recipes/proxy.rb

```diff
@@ -20,6 +20,18 @@
   content '<h1>WORKS!</h1>'
 end
 
+execute 'generate resolver config' do
+  command <<-EOH
+    echo resolver $(awk 'BEGIN{ORS=" "} /nameserver/{print $2}' /etc/resolv.conf | sed "s/ $/;/g") >/etc/nginx/conf.d/resolver.conf
+  EOH
+  notifies :reload, 'service[nginx]'
+end
+
```

Comment:
Nice :) In other places we've just put hard-coded the google servers, eg 8.8.8.8. Is it worth factoring this logic into a common cookbook to share across others?